src/battle_main.c

```diff
@@ -1936,8 +2212,21 @@ u8 CreateNPCTrainerPartyFromTrainer(struct Pokemon *party, const struct Trainer
             monsCount = trainer->partySize;
         }
 
+        bool32 usingPool = FALSE;
+        u8 *poolIndexArray;
```

Comment:
I think this should be initialized to NULL. If nothing else it helps people who may still have `-werror=maybe-uninitialized` on, or who want to turn it on.

Reply:
Isn't it already on?
If I try to add a usage of it that I know is uninitialized it will warn me, but not with the current setup.

Reply:
![20241205_20h27m01s_grim](https://github.com/user-attachments/assets/b55c587c-5a77-425a-b42d-f15d7b83c464)

This is if I add a known uninitialized usage, the compiler seems smart enough to see that it can't be used uninitialized currently because its usage depends on a value being true that's set in the same function it's initialized in.

Reply:
Weird. Perhaps it was my version of gcc, but I was getting `-Werror=maybe-uninitialized` with the current setup.

Reply:
gcc (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0 btw

Reply:
gcc (GCC) 14.2.1 20240910
Does not give an error.

Reply:
> Weird. Perhaps it was my version of gcc, but I was getting `-Werror=maybe-uninitialized` with the current setup.

It probably is—GCC gets more advanced static analysis with every version :)

~~SBird will swing by in a minute to say that we should ensure we support older GCCs, like in #5611~~

Reply:
There's no harm in initializing early, so may as well

Reply:
Added the `NULL`.
(And some other fixes)